distributed/shuffle/_shuffle_extension.py

```diff
@@ -378,7 +435,9 @@ async def _barrier(self, shuffle_id: ShuffleId) -> None:
         await shuffle.barrier()
 
     async def _register_complete(self, shuffle: Shuffle) -> None:
+        self.raise_if_closed()
         await shuffle.close()
+        self.raise_if_closed()
```

Comment:
These two raise_if_closed also seem to have no effect. I understand that at every tick we might've received a message from the scheduler setting an exception but it should not be necessary to check at every stop, is it?

Reply:
These two can be removed which would subtly change the semantics of when a shuffle is allowed to succeed. With them, a shuffle cannot register as complete if we have closed the extension since we got the last output partition. The shuffle is still performing work and the extension is not allowed to perform any more work. Without them, a shuffle is allowed to succeed as long as we manage to get all output partitions before the extension closes. We got all the important work done and now we're just telling the scheduler about it. Both semantics seem fine to me.

I'd argue that the change won't even make a difference in most cases because it's quite likely that either the extension closed because a worker left, so it is already doomed, or the worker will proceed to shut down, which might release the output partition data. 

In either case, the first one is not strictly needed since `Shuffle.close` is idempotent by now and guaranteed to wait until the shuffle is completely closed.

Reply:
Dropped both of these for the sake of simplicity and added a comment about why no check is needed.